gammapy/image/core.py

```diff
@@ -444,27 +447,124 @@ def pad(self, mode, pad_to_factor=None, pad_width=None, **kwargs):
-                             'either "pad_width" or "pad_to_factor"')
+                             'either "pad_width", "pad_to_factor" or "shape"')
+        if args_none.sum() < 2:
+            raise ValueError('Indicate only one parameter: '
+                             'either "pad_width", "pad_to_factor" or "shape"')
 
         if pad_to_factor is not None:
             pad_width = pad_to_factor - (np.array(self.data.shape) % pad_to_factor)
             pad_width = [(0, pad_width[0]), (0, pad_width[1])]
 
-        # converting from unicode to ascii string as a workaround
-        # for https://github.com/numpy/numpy/issues/7112
-        mode = str(mode)
+        if shape is not None:
+            xdiff = shape[1] - self.data.shape[1]
+            ydiff = shape[0] - self.data.shape[0]
+
+            if (np.array([xdiff, ydiff]) % 2).any():
+                raise ValueError('For symmetric padding, difference to new shape '
+                                 'must be even in all axes.')
+
+            xpad, ypad = xdiff // 2, ydiff // 2
+            pad_width = ((ypad, ypad), (xpad, xpad))
+            wcs.wcs.crpix += np.array([xpad, ypad])
 
         data = np.pad(self.data, pad_width=pad_width, mode=mode, **kwargs)
+        return SkyImage(data=data, wcs=wcs)
+
+    def crop(self, shape=None):
+        """
+        Crop sky image symmetrically to a given shape.
+
+        Paramters
+        ---------
+        shape : tuple
+            Desired shape.
+        """
+        xdiff = (self.data.shape[1] - shape[1])
+        ydiff = (self.data.shape[0] - shape[0])
+
+        if (np.array([xdiff, ydiff]) % 2).any():
+            raise ValueError('For symmetric cropping, difference to new shape '
+                             'must be even in all axes.')
+
+        x_crop = xdiff // 2
+        y_crop = ydiff // 2
+        data = self.data[y_crop:-y_crop, x_crop:-x_crop]
+
+        wcs = self.wcs.deepcopy()
+        wcs.wcs.crpix -= np.array([x_crop, y_crop])
+        return SkyImage(data=data, wcs=wcs)
+
+    def downsample(self, factor, method=np.nansum):
+        """
+        Down sample image by a given factor.
+
+        The image is down sampled using `skimage.measure.block_reduce`. If the
+        shape of the data is not divisible by the down sampling factor, the image
+        must be padded beforehand to the correct shape.
 
-        # We don't have to adjust WCS here, because we only pad on the
-        # right and top, and for this change, the CRPIX doesn't change.
+        Parameters
+        ----------
+        factor : int
+            Down sampling factor, must be power of two.
+        method : np.ufunc (np.nansum), optional
+            Method how to combine the image blocks.
+
+        Returns
+        -------
+        image : `SkyImage`
+            Down sampled image.
+        """
+        from skimage.measure import block_reduce
+
+        shape = self.data.shape
+        factor = int(factor)
```

Comment:
The doc says `factor` has to be int.
I'm -1 an auto-truncating floats by calling `factor = int(factor)` here.
Either remove this line, expecting the user to follow the docs.
Or if you want to ensure int, instead of silently truncating, I'd suggest throwing an error if the input isn't an int.